### ISSUE TITLE
changelog: v76-beta

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#940]
* date: 7-May-2020
* foreword: These are all the changes since the last changelog ([v75-beta](https://github.com/ghacksuserjs/ghacks-user.js/issues/928)).

[FF76 Release notes](https://www.mozilla.org/en-US/firefox/76.0/releasenotes/)
[FF76 for developers](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/76)
[FF76 compatibility](https://www.fxsitecompat.com/en-CA/versions/76/)
[FF76 security advisories](https://www.mozilla.org/en-US/security/advisories/mfsa2020-16/)

---

#### changelog: [[all changes](https://github.com/ghacksuserjs/ghacks-user.js/compare/v75.0-beta...v76.0-beta)]
* all user.js **updates for Firefox v76** are detailed in the [ToDo: diffs FF75-FF76](https://github.com/ghacksuserjs/ghacks-user.js/issues/932) issue
  * includes links to the commits made for each pref, links to bugzilla tickets, our discussions etc.
* for all the rest see the full list of pref changes below

---

#### all pref changes:

* new in 76beta but commented out by default
```js
//user_pref("dom.security.https_only_mode", true);
```

* moved to `9999: DEPRECATED / REMOVED` :
```js
user_pref("extensions.blocklist.url", "https://blocklists.settings.services.mozilla.com/v1/blocklist/3/%APP_ID%/%APP_VERSION%/");
```

---

Any and all help, suggestions, recommendations, links, tips and tricks, questions, thank you's or what have you are welcome - signup/login and start typing - thanks

**Comment by iWARR** [comment]
**ExperimentManager & API**

ExperimentManager module and API in order to support Messaging System Experiments / AboutWelcome. Handles Branch generation, Caching, Telemtry and other elements of experiment lifecycle management. Can be integrated with MS and Normandy.

**Features:**	
•  Expose lifecycle methods for handling new recipes, ready states
•  Work at startup and first run
•  Provide an API that allows consumer code (incl. MessagingSystemRouter and about:welcome)

[1]		https://bugzilla.mozilla.org/show_bug.cgi?id=1620021

```
// ExperimentManager & API =  [ Disabled ]
user_pref("messaging-system.rsexperimentloader.enabled", false);	// [FF76]
user_pref("messaging-system.log", "");					// [DEFAULT = "Warn"] // [FF76]
```

```
//  about:welcome - Dynamic triplets = [ Disabled ]
/*	Simplified UI version of the "about:welcome". 
	about:home & about:newtab are also affected and have dependency on analysis from experiment.
[1]	https://github.com/ghacksuserjs/ghacks-user.js/issues/932     */
user_pref("trailhead.firstrun.branches", "");					// [DEFAULT = "join-dynamic"] // [FF76]
```

**Comment by iWARR** [comment]
Also I have in my own diffs:
- 4 unwanted prefs relative to password management (I don't meet the new Mozilla's intrusion here - any "breached checks" and OS auth), including logging. I think these "improvements" more than redudant, because we already have "Master password" option for those who need it. I'm highly dislike connecting OS Windows (weak and untrusted itself) auth into the browser. 
- 2 not bad GFX options that can improve performance, so I put them enabled

I'm afraid these prefs are outside of "ghacks-user.js" coverage, so I'm going to stay out of posting this for a while.

**Comment by Thorin-Oakenpants** [comment]
What is your point? Did the word "experiment" cause confirmation bias and spook you? Can you show me proof there is a privacy etc issue with this API? AFAIK, the API is so they have central message handling - rather than per experiment / per prefs..Experiments / normandy are already covered under the master telemetry switch in `0340` (`datareporting.healthreport.uploadEnabled`) as well as `0342` (`app.shield.optoutstudies.enabled`) for a little in depth defense. Both of which have a UI presence in about:preferences#privacy

As for anything `firstrun` - what's the point? Are you like creating new profiles for every Firefox session? How do you stop it when that profile hasn't been created yet (it can be done but is outside the scope of this repo)? Can you show me where in trailhead there is a privacy, tracking, security, fingerprinting, or anonymity problem?

If you have set a master password (as recommended by the user.js - there's no pref for it - see `0902`), then the OS integration is not used. Otherwise, as a default, it's good to have it integrated. Any other password prefs you have issues with: again: show me where it's a bad thing regards privacy, security, etc

**Comment by iWARR** [comment]
1) Active Stream, Telemetry, Normandy, Built-In trash plugins and tons of other bloatware are well known. Yesterday - default-browser-agent.exe in the Sheduler, today - ExperimentManager, a new same "feature". I'm always kill such stuff, like many-many others.
Sometimes it's hard to understand were the "Red-Line" of "ghacks-user.js" is...

Can you aswer me why all this stupid-kid games with a simple Auto/Update option, not too different like hunderds others? Why restricting this policy to ESR-only? At least, why I should hunting now for this here:

C:\Program Files\Mozilla Firefox\application.ini

2) OS Windows auth protection is weak itself. You need to use "Autorun" to encrypt OS plain password, at least. Why spreading a tons of Windows vulnerabilities into the browser? Until they were separated, I kept calm. The true is as less untrusted OS knows about (not very)trusted browser, as it's better.

I know you may mark this as "oftopic", but that doesn't change the heart of the truth.

**Comment by Thorin-Oakenpants** [comment]
- Auto updating can already be disabled, by pref (which we do) and also in the UI options. It is only the update **check** that was restricted. If you want a definitive answer as to why Mozilla did that, then read the bugzillas - but it's not "stupid-kid games"
- Password: I don't fucking care - I already told everyone to set their own in the user.js. If you don't set your own master password, then at least the Windows one is better than nothing - but not my fucking problem

**Comment by iWARR** [comment]
My closing words are the next: the browser have to stay "isolated" from the untrusted OS as much as possible.
Or we will read sad infosec news and there will be too late for Mozilla's apologise.

Did you noticed that Mozilla is too involved into the "user passwords" questions last time?
Temporary emails on Mozillas servers, breached checks, and now dealing with OS passwords.
Come closer to me, baby... It doesn't hurt.
What the next?

Of course, all of these are for your "well-being".
Google's SafeBrowsing uses partial hashes for your checks, Mozilla uses the same for your passwords by default.

Are you are scared? Or not? It's doesn't matter, while you are ... CHECKED!

(Yes. It's not anybody f@ing problem, except only YOU (who read this)).

**Comment by Thorin-Oakenpants** [comment]
^^ overworked... (and underpaid) as per usual...  I fucked up the label ... took me 5 days to notice .. see what you guys are doing to me .. just slow down

---

FYI: [1599138](https://bugzilla.mozilla.org/show_bug.cgi?id=1599138) .. comment 12 is informative

5000's personal section
```
/* UX FEATURES: disable and hide the icons and menus ***/
   // user_pref("browser.messaging-system.whatsNewPanel.enabled", false); // What's New [FF69+]
   // three more prefs listed
```

I've known for ages that the pref doesn't actually hide the menu (but I don't think it's worth pointing that out for that one pref in the bunch of four under that sub-title). Tested just now in nightly and the menu is still there (I mention this because my menu item is missing - see below, further down .. no, further .. a bit more .. there you go)

I know that for several releases after 70 (which is when it took effect: see [1563319](https://bugzilla.mozilla.org/show_bug.cgi?id=1563319) I still saw the gift icon in the nav bar after updates. IDK if all major releases have changes to what's new and this is what triggers the badging/icon/panel or whatever.

But I haven't seen one for a long time: and, to my surprise, when I just checked now, my menu item is not there (and it's not userContent css either).
  - ❓  **I'm not sure why I don't have one, TBH. If anyone knows, fill me in** 

Anyway: there is a new SETTING (but not worth adding) in FF78+. Hamburger menu > What's New and at the bottom is a checkbox for "Notify about new features" - this is the setting for the pref. Clearly as you can see it's disabled, but the menu item is still there (by design: see comment 12, how else could someone toggle it back on, for example)

![wotsnew](https://user-images.githubusercontent.com/16656956/81651574-4c697500-9422-11ea-92ae-ecd4d9fa8c4e.png)

**Comment by crssi** [comment]
> I'm not sure why I don't have one, TBH. If anyone knows, fill me in

When switching `browser.messaging-system.whatsNewPanel.enabled` to `false`, the "Whats new" went gone from the hamburger menu and still missing even when `browser.messaging-system.whatsNewPanel.enabled` is now set back to `true`

**Comment by crssi** [comment]
^^ until reopening FF

**Comment by Thorin-Oakenpants** [comment]
OK, I worked it out. The menu change is only in 78+

My test suite of portable FF's all have a basic user.js to save me configuring things when I set each new one: nothing major: just things like turning off AS, pocket, screenshots, sync, lasttab, warn on close, reader view, recommendations, etc. Included in that is the what's new pref.

I don't know if it requires a restart (that bugzilla mentions reading the pref early in startup) or some prefs observer detects it prior to 78, but in 78+ it at least recognizes the change via the new UI (but not necessarily via pref changing or user.js except on startup)

Anyway: the menu is gone prior to 78 (tested in beta 77, dev 77, 76, 75 - i.e tested as started with that, not toggled via about:config), but in 78 it sticks around (so the UI can be used). I haven't seen the menu or nav bar icon since probably 74 (or even 73) so I don't what all those bugzillas (see all those marked as duplicates) are on about exactly (maybe they're all old and relate to 73/74 but I'm not going to go read em all) and the fix in 78 was resolve the appearing/disappearing menu. IDK. Don't care .. Just thought I'd share in case someone else came across it in 78
